lib/common.js

```diff
@@ -792,6 +791,37 @@ function getSharkInfo(req, hostname) {
     return (sharks[0]);
 }
 
+/**
+ * Given an resource like `/mark/stor/my%resume.pdf`, this API will convert to
```

Comment:
I haven't had a chance to dig in but, I have this sneaking suspicion that this "normalization" is just wrong, at least for buckets API.  Buckets API should *not* be blindly running `decodeURIComponent` on every slash-separated component of the full URL path.  Its individual endpoints should be properly handling encoding/decoding of the separate `login`, `bucketName`, and `objectName` fields, no?

I'm fine if looking into that is separate from this ticket. I gather this is just about moving existing usage over from libmanta.

Reply:
I haven't looked into this at all - my goal was just to pull `libmanta` into this file like you mentioned.  I'll create a separate ticket for this.

Reply:
https://jira.joyent.us/browse/MANTA-4780